Src/FluentAssertions/Primitives/NullableBooleanAssertions.cs

```diff
@@ -27,6 +27,11 @@ public NullableBooleanAssertions(bool? value)
         {
         }
 
+        /// <summary>
+        /// Gets the object which value is being asserted.
+        /// </summary>
+        public new bool? Subject => SubjectInternal;
```

Comment:
I don't get what you're trying to solve here.

Reply:
I wanted to change the type of the property `Subject`, such that it's
* `bool` for `BooleanAssertions`, and
* `bool?` for `NullableBooleanAssertions`.

Reply:
It's ugly, but I don't see an alternative solution either.